spec/models/reservation_spec.rb

```diff
@@ -21,6 +21,72 @@
   # can't run if nil (?)
   #
 
+  describe 'counter cache' do
+    context 'newly overdue reservation' do
+      it 'increments when a reservation is marked as overdue' do
+        model = FactoryGirl.build_stubbed(:equipment_model)
+        allow(EquipmentModel).to receive(:find).with(model.id).and_return(model)
+        res = FactoryGirl.create(:checked_out_reservation,
+                                 equipment_model: model,
+                                 start_date: Time.zone.today - 2.days,
+                                 due_date: Time.zone.today - 1.day)
+        res.update_columns(overdue: false)
+        expect(model).to receive(:increment).with(:overdue_count).once
+        res.update_attributes(overdue: true)
+      end
+    end
+    context 'already overdue reservation' do
+      it "updating other attributes doesn't affect the cache" do
+        model = FactoryGirl.build_stubbed(:equipment_model)
+        allow(EquipmentModel).to receive(:find).with(model.id).and_return(model)
+        res = FactoryGirl.create(:overdue_reservation, equipment_model: model)
+        expect(model).not_to receive(:increment)
+        res.update_attribute(:notes, 'test')
+      end
+      it 'decrements when checked in' do
+        model = FactoryGirl.build_stubbed(:equipment_model)
+        allow(EquipmentModel).to receive(:find).with(model.id).and_return(model)
+        res = FactoryGirl.create(:overdue_reservation, equipment_model: model)
+        expect(model).to receive(:decrement).with(:overdue_count).once
+        res.update_attributes(
```

Comment:
I wish we could wrap this in a nice instance method (e.g. `checkin`) instead that didn't require a whole bunch of other setup (mocking users, etc) - that would be a nice improvement for the Reservation model (I'm surprised we don't have an open issue for that). I could also imagine creating a service object for it (i.e. `ReservationCheckerIn`, what a terrible name), and then I guess this spec would go in there instead.